packages/dev/core/src/Loading/Plugins/babylonFileLoader.ts

```diff
@@ -959,25 +959,36 @@ SceneLoader.RegisterPlugin({
             }
 
             // Fog
-            if (parsedData.fogMode && parsedData.fogMode !== 0) {
+            if (parsedData.fogMode) {
                 scene.fogMode = parsedData.fogMode;
+            }
+            if (scene.fogColor) {

```

Comment:
I think you mean to test on `parsedData`, not `scene`?

Also, I think it's better to test against `undefined` because 0 is a valid value for some of the properties.

Same comment for all checks below.

Reply:
yes lol